clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientFutureImpl.java

```diff
@@ -90,7 +92,16 @@ private RuntimeException transformExecutionException(final ExecutionException e)
 
     if (cause instanceof StatusRuntimeException) {
       final Status status = ((StatusRuntimeException) cause).getStatus();
-      throw new ClientStatusException(status, e);
+      if (status.getCode() == Code.UNAVAILABLE) {
+        final Throwable statusCause = status.getCause();
+        if (statusCause == null) {
+          throw new GatewayConnectionException(cause);
+        } else {
+          throw new GatewayConnectionException(statusCause);
+        }
+      } else {
+        throw new ClientStatusException(status, e);
+      }
```

Comment:
I'm not sure I understand these changes. When do we expect to use `status.getCause()`, and when not? What would be the different cases/use cases? Why not simply pass the cause immediately?

Reply:
Ah, I think I see - the solution is clever, but doesn't seem very portable/future proof. If the stack trace ever changes (even just the order), then the message may become difficult to understand again. I like that we add a better message, but I'm unsure if pulling the cause up helps us at all.

Maybe we can figure out a better way to get the underlying cause? I guess that's something the gRPC library would know better than we do. I can do a bit of digging see what others do with this